src/analyze/accumulatedtracedata.cpp

```diff
@@ -56,6 +58,44 @@ ostream& operator<<(ostream& out, const Index<Base> index)
     out << index.index;
     return out;
 }
+
+// boost's counter filter uses an int for the count which overflows for large streams; so replace it with a work alike.
+class byte_counter {
+public:
+    using char_type = char;
+    struct category
+        : boost::iostreams::dual_use
+        , boost::iostreams::filter_tag
+        , boost::iostreams::multichar_tag
+        , boost::iostreams::optimally_buffered_tag
+    {};
+
+    uint64_t bytes() const { return bytes_; }
+    std::streamsize optimal_buffer_size() const { return 0; }
+
+    template<typename Source>
+    std::streamsize read(Source& src, char* str, std::streamsize size)
+    {
+        auto const readsize = boost::iostreams::read(src, str, size);
+        if (readsize == -1)
+            return -1;
+        bytes_ += readsize;
+        return readsize;
+    }
+
+    template<typename Sink>
+    std::streamsize write(Sink& sink, const char* str, std::streamsize size)
+    {
+        auto const writesize = boost::iostreams::write(sink, str, size);
+        bytes_ += writesize;
+        return writesize;
+    }
+
+private:
+    uint64_t bytes_ = 0;
```

Comment:
`m_bytes` is the format used in heaptrack

Reply:
done.